File: LibraryMe.API/BookLibrary.DAL/Models/DTO/ReservationDTO.cs
namespace BookLibrary.DAL.Models.DTO
{
    public class ReservationDTO
    {
        public int ReservationId { get; set; }
        public DateTime CreatedTime { get; set; }
        public DateTime? AcceptedTime { get; set; }
        public DateTime? CheckOutTime { get; set; }
        public string Status { get; set; }
        public  IEnumerable<BookShortcutDTO> Books { get; set; }
    }
}

File: LibraryMe.UI/LibraryMe/src/app/Features/Reservations/Models/reservation-dto.ts
import {BookShortcutDTO} from "../../Books/models/book-shortcut-dto";

export interface ReservationDTO {
    reservationId: number;
    createdTime: string; // Adjust the type based on your DateTime format
    acceptedTime?: string | null; // Adjust the type based on your DateTime format
    checkOutTime?: string | null; // Adjust the type based on your DateTime format
    status: string;
    books: BookShortcutDTO[];
}
